torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -244,6 +244,25 @@ std::vector<Shape> compute_shape_min(const at::Tensor & self){
     return {Shape(self.scalar_type(), {})};
 }
 
+std::vector<Shape> compute_shape_nonzero(const at::Tensor& t, bool as_tuple) {
+  if (as_tuple) {
+    auto res = std::vector<Shape>();
+    for (auto dim_size : t.sizes()) {
+      res.emplace_back(Shape(at::kLong, {dim_size}));
+    }
+    return res;
+  }
+  int64_t max_elements = 1;
+  for (auto dim_size : t.sizes()) {
+    max_elements *= dim_size;
+  }
+  return {Shape(at::kLong, {max_elements, (int64_t)t.sizes().size()})};
```

Comment:
Please help me understand why we don't do something like this:

`return {Shape(at::kLong, std::vector<int64_t>(t.sizes().begin(), t.sizes().end()))};`